modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -325,24 +340,51 @@ impl<T: Config<I>, I: 'static> SendXcm for Pallet<T, I> {
 			return Err(SendError::ExceedsMaxMessageSize)
 		}
 
+		// compute bridge id from universal locations
+		let bridge_destination_relative_location =
+			dest.as_ref().map(|dest| dest.clone()).ok_or(SendError::MissingArgument)?;
+		let bridge_origin_universal_location = T::UniversalLocation::get();
+		let bridge_destination_universal_location: InteriorMultiLocation =
+			bridge_origin_universal_location
+				.into_location()
+				.appended_with(bridge_destination_relative_location)
+				.map_err(|_| SendError::NotApplicable)?
+				.try_into()
+				.map_err(|_| SendError::NotApplicable)?;
+		let bridge_id = bridge_id_from_locations(
+			&T::UniversalLocation::get(),
+			&bridge_destination_universal_location,
+		);
+
 		// just use exporter to validate destination and insert instructions to pay message fee
 		// at the sibling/child bridge hub
 		//
 		// the cost will include both cost of: (1) to-sibling bridg hub delivery (returned by
 		// the `Config::ToBridgeHubSender`) and (2) to-bridged bridge hub delivery (returned by
 		// `Self::exporter_for`)
 		ViaBridgeHubExporter::<T, I>::validate(dest, xcm)
-			.map(|(ticket, cost)| ((message_size, ticket), cost))
+			.map(|(ticket, cost)| ((bridge_id, message_size, ticket), cost))
 	}
 
 	fn deliver(ticket: Self::Ticket) -> Result<XcmHash, SendError> {
 		// use router to enqueue message to the sibling/child bridge hub. This also should handle
 		// payment for passing through this queue.
-		let (message_size, ticket) = ticket;
-		let xcm_hash = ViaBridgeHubExporter::<T, I>::deliver(ticket)?;
+		let (bridge_id, message_size, ticket) = ticket;
+		let bridge = Bridges::<T, I>::get(bridge_id).ok_or_else(|| SendError::Transport("UnknownBridge"))?;
+		let xcm_hash = if bridge.is_suspended() {
+			let xcm_hash = Default::default(); // TODO: what about message hash? is it important? where it is used?
+			let message_index = LatestSuspendedMessageIndex::<T, I>::mutate(|index| {
+				*index += 1;
+				*index
+			});
+			SuspendedMessages::<T, I>::insert(message_index, ticket.encode().try_into().expect("TODO"));
```

Comment:
If the bridge is suspended (because some of its channels are congested), we have 2 options here imo:
1) save messages here and send them later, when bridge operations are resumed - that's what I'm trying to implement here;
2) (the easiest one) simply return an error. Then the sender will get a corresponding error and his transaction will fail. So tokens are not lost (given that the "delivery" is called by actual sender - if it's just a call somewhere in the middle of the path, it'll be lost)

WDYT is better?

Reply:
hmm.. the first option makes using this pallet easier - no need to verify all paths leading to this pallet and whether messages/tokens can be lost, and the implementation here doesn't add much complexity so I say we stay with 1.

We'll ultimately see in audit 🙈 😆